hardware/Meteorologisk.cpp

```diff
@@ -0,0 +1,446 @@
+	{
+		wind_degrees = static_cast<float>(atof(instantData["wind_from_direction"].asString().c_str()));
+	}
+	if (instantData["wind_speed"].empty()==false)
+	{
+		windspeed_ms = static_cast<float>(atof(instantData["wind_speed"].asString().c_str()));
+	}
+	if (instantData["wind_speed_of_gust"].empty()==false)
+	{
+		windgust_ms = static_cast<float>(atof(instantData["wind_speed_of_gust"].asString().c_str()));
+	}
+	if (wind_degrees!=-1)
+	{
+		RBUF tsen;
+		memset(&tsen,0,sizeof(RBUF));
+		tsen.WIND.packetlength=sizeof(tsen.WIND)-1;
+		tsen.WIND.packettype=pTypeWIND;
+		tsen.WIND.subtype=sTypeWIND4;
+		tsen.WIND.battery_level=9;
+		tsen.WIND.rssi=12;
+		tsen.WIND.id1=0;
+		tsen.WIND.id2=1;
+
+		float winddir=float(wind_degrees);
+		int aw=round(winddir);
+		tsen.WIND.directionh=(BYTE)(aw/256);
+		aw-=(tsen.WIND.directionh*256);
+		tsen.WIND.directionl=(BYTE)(aw);
+
+		tsen.WIND.av_speedh=0;
+		tsen.WIND.av_speedl=0;
+		int sw=round(windspeed_ms*10.0f);
+		tsen.WIND.av_speedh=(BYTE)(sw/256);
+		sw-=(tsen.WIND.av_speedh*256);
+		tsen.WIND.av_speedl=(BYTE)(sw);
+
+		tsen.WIND.gusth=0;
+		tsen.WIND.gustl=0;
+		int gw=round(windgust_ms*10.0f);
+		tsen.WIND.gusth=(BYTE)(gw/256);
+		gw-=(tsen.WIND.gusth*256);
+		tsen.WIND.gustl=(BYTE)(gw);
+
+		//this is not correct, why no wind temperature? and only chill?
+		tsen.WIND.chillh=0;
+		tsen.WIND.chilll=0;
+		tsen.WIND.temperatureh=0;
+		tsen.WIND.temperaturel=0;
+
+		tsen.WIND.tempsign=(wind_temp>=0)?0:1;
+		int at10=round(std::abs(wind_temp*10.0f));
+		tsen.WIND.temperatureh=(BYTE)(at10/256);
+		at10-=(tsen.WIND.temperatureh*256);
+		tsen.WIND.temperaturel=(BYTE)(at10);
+
+		tsen.WIND.chillsign=(wind_chill>=0)?0:1;
+		at10=round(std::abs(wind_chill*10.0f));
+		tsen.WIND.chillh=(BYTE)(at10/256);
+		at10-=(tsen.WIND.chillh*256);
+		tsen.WIND.chilll=(BYTE)(at10);
+
+		sDecodeRXMessage(this, (const unsigned char *)&tsen.WIND, NULL, 255);
+	}
+
+	//UV
+	if (instantData["ultraviolet_index_clear_sky"].empty() == false)
+	{
+		float UV = instantData["ultraviolet_index_clear_sky"].asFloat();
+		if ((UV < 16) && (UV >= 0))
+		{
+			SendUVSensor(0, 1, 255, UV, "UV Index");
+		}
+	}
+
+	//Rain
+	if (root["currently"]["precipIntensity"].empty()==false)
+	{
+		if ((root["currently"]["precipIntensity"] != "N/A") && (root["currently"]["precipIntensity"] != "--"))
+		{
+			float rainrateph = static_cast<float>(atof(root["currently"]["precipIntensity"].asString().c_str()))*25.4f; //inches to mm
+			if ((rainrateph !=-9999.00f)&&(rainrateph >=0.00f))
+			{
+				SendRainRateSensor(1, 255, rainrateph, "Rain");
+			}
+		}
+	}
+
+	//Visibility
+	if (root["currently"]["visibility"].empty()==false)
+	{
+		if ((root["currently"]["visibility"] != "N/A") && (root["currently"]["visibility"] != "--"))
+		{
+			float visibility = static_cast<float>(atof(root["currently"]["visibility"].asString().c_str()))*1.60934f; //miles to km
+			if (visibility>=0)
+			{
+				_tGeneralDevice gdevice;
```

Comment:
Please use CDomoticzHardwareBase::SendVisibilitySensor

Reply:
Just removed it as I do not see it in Locationforecast either, same for radiation